http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -58,72 +59,33 @@
         }
     }
 
-    private final LoginContext loginContext;
-    private final GSSCredential clientCredential;
+    private final File keytab;
+    private final File credentialCache;
+    private final String principal;
     private final String remoteServiceName;
+    private LoginContext loginContext;
```

Comment:
This class may be accessed concurrently by multiple threads, so these fields need to be properly synchonized